docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,716 @@
+    #
+    # Example for Alicloud:
+    #
+    # apiVersion: alicloud.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   id: coreos_2023_4_0_64_30G_alibase_20190319.vhd
+    #
+    #
+    # Example for AWS:
+    #
+    # apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   regions:
+    #   - name: europe-central-1
+    #     ami: ami-0f46c2ed46d8157aa
+    #
+    #
+    # Example for Azure:
+    #
+    # apiVersion: azure.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   publisher: CoreOS
+    #   offer: CoreOS
+    #   sku: Stable
+    # countFaultDomains:
+    # - region: westeurope
+    #   count: 2
+    # countUpdateDomains:
+    # - region: westeurope
+    #   count: 5
+    #
+    #
+    # Example for GCP:
+    #
+    # apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: projects/coreos-cloud/global/images/coreos-stable-2023-5-0-v20190312
+    #
+    #
+    # Example for OpenStack:
+    #
+    # apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: coreos-2023.5.0
+    # keyStoneURL: https://url-to-keystone/v3/
+    # dnsServers:
+    # - 10.10.10.10
+    # - 10.10.10.11
+    # dhcpDomain: foo.bar
+    # requestTimeout: 30s
+    # constraints:
+    #   loadBalancerProviders:
+    #   - name: haproxy
+    #   floatingPools:
+    #   - name: fip1
+    #     loadBalancerClasses:
+    #     - name: class1
+    #       floatingSubnetID: 04eed401-f85f-4610-8041-c4835c4beea6
+    #       floatingNetworkID: 23949a30-1cdd-4732-ba47-d03ced950acc
+    #       subnetID: ac46c204-9d0d-4a4c-a90d-afefe40cfc35
+    #
+    #
+    # Example for Packet:
+    #
+    # apiVersion: packet.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2079.3.0
+    #   id: d61c3912-8422-4daf-835e-854efa0062e4
+```
+
+### `Seed` resource
+
+Special note: The proposal contains field that are not yet existing in the current `garden.sapcloud.io/v1beta1.Seed` resource, but they should be implemented (open issues that require them are linked).
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seed-secret
+  namespace: garden
+type: Opaque
+data:
+  kubeconfig: base64(kubeconfig-for-seed-cluster)
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Seed
+metadata:
+  name: seed1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
```

Comment:
Why is the `region` needed? 
If this is the region where the Seed is hosted - how about to fetch the region just like it is done for the plant resource and put it into the status?
Or this is used for some limitation of features that are (not) available in particular region?

Reply:
The region is needed for the `gardener-scheduler`. It always tries to place a shoot's control plane into a seed that is closest (if shoot is in region `eu` it tries to find a seed in region `eu`). Depending on the configured strategy, only if it cannot find such seed it chooses one in another region.

Fetching it from the cluster directly might not work in all cases (actually, it might similarly not always work for `Plant`s). The reason is that we don't know how the seed cluster was provisioned (which tool was used). The `Plant` controller fetches the region from the node labels, but if these labels don't exist then it never knows the region. Thinking about e.g. bare metal providers with their own CCMs, etc., I doubt that all of them implement this label.
So, I think it's still safer to let the administrator explicitly define the region (needs to be done anyways only once during initial registration).

Reply:
Thanks, I got it.